shared/src/main/kotlin/com/egm/stellio/shared/model/ExpandedEntity.kt

```diff
@@ -86,27 +89,47 @@ data class ExpandedEntity(
             this.plus(propertyKey to JsonLdUtils.buildNonReifiedTemporalValue(dateTime))
         else this
 
-    fun filterOnAttributes(includedAttributes: Set<String>): Map<String, Any> {
+    fun filterOnAttributes(includedAttributes: Set<String>, includedDatasetIds: Set<URI>?): Map<String, Any> {
         val inputToMap = { i: ExpandedEntity -> i.members }
-        return filterEntityOnAttributes(this, inputToMap, includedAttributes)
+        return filterEntityOnAttributes(this, inputToMap, includedAttributes, includedDatasetIds)
     }
 
-    private fun <T> filterEntityOnAttributes(
-        input: T,
-        inputToMap: (T) -> Map<String, Any>,
-        includedAttributes: Set<String>
+    private fun filterEntityOnAttributes(
+
+        input: ExpandedEntity,
+        inputToMap: (ExpandedEntity) -> Map<String, Any>,
+        includedAttributes: Set<String>,
+        includedDatasetIds: Set<URI>?,
     ): Map<String, Any> {
-        return if (includedAttributes.isEmpty()) {
-            inputToMap(input)
-        } else {
+        var result = inputToMap(input)
```

Comment:
avoid `var` (use `val` instead)

Reply:
it is modified, i suggested this implementation to have each filter apply their change independently.

Reply:
disclaimer: raw and untested

```
        return result.filterKeys {
            val includedKeys = JSONLD_EXPANDED_ENTITY_CORE_MEMBERS.plus(includedAttributes)
            includedKeys.contains(it)
        }.mapNotNull { entry ->
            if (entry.key in JSONLD_EXPANDED_ENTITY_CORE_MEMBERS)
                entry.value
            else (entry as ExpandedAttributes).mapValues { (_, expandedAttributeInstances) ->
                expandedAttributeInstances.filter { expandedAttributeInstance ->
                    // TODO : what if the default datasetId is asked for
                    //  btw, how do we ask for the defaut datasetId?
                    includedDatasetIds.isNullOrEmpty() ||
                        includedDatasetIds.contains(expandedAttributeInstance.getDatasetId())
                }.ifEmpty { null }
            }
        }
```

Reply:
replying to my TODO: "The datasetId of the default Attribute instance can be specified as "@none" (4.5.5)